### ISSUE TITLE
regarding iterations within each epoch and dropout setting

### ISSUE THREAD

**Issue by wenouyang** [jocicmarko/ultrasound-nerve-segmentation#11]
Hi Marko,

Regarding the the main optimization step `model.fit(imgs_train, imgs_mask_train, batch_size=32, nb_epoch=20, verbose=1, shuffle=True,
              callbacks=[model_checkpoint])`

How many iterations within each epoch will it be included? Here, we have about 5635 pairs of training images and training mask images. If batch_size=32, should it be 5635/32 iterations?
I did not find a place where this parameter of training_iters are setup. Will it be handled automatically by model.fit?

Besides, how do you setup the dropout rate in this training process? Will it be helpful?

Thank you very much.

**Comment by jocicmarko** [comment]
Hi,

You are correct, Keras ```model.fit()``` handles the number of iterations automatically (based on number of epochs, size of the training set and batch size).

Dropout can be added with ```Dropout``` layer (check https://keras.io/layers/core/#dropout), the use is the same like any other layer. Vanilla dropout after convolutional layers is not something that is often done in practice, some papers suggest using spatial dropout, available in Keras via ```SpatialDropout2D``` layer. I haven't used it for this particular problem though.

Hope this helps,
Marko

**Comment by wenouyang** [comment]
Hi Marko,

Thanks for the reply.
